src/foam/u2/view/RichChoiceView.js

```diff
@@ -264,26 +266,29 @@ foam.CLASS({
           }))
           .add(this.slot(function(sections) {
             return this.E().forEach(sections, function(section) {
-              this
-                .start()
-                  .addClass(self.myClass('heading'))
-                  .add(section.heading)
-                .end()
-                .start()
-                  .select(section.filtered || section.dao, function(obj) {
-                    return this.E()
-                      .start(self.rowView, { data: obj })
-                        .enableClass('disabled', section.disabled)
-                        .callIf(! section.disabled, function() {
-                          this.on('click', () => {
-                            self.fullObject_ = obj;
-                            self.data = obj;
-                            self.isOpen_ = false;
+              var positionElement = this;
+              section.dao.select().then((resp) => {
```

Comment:
It would be better to use a COUNT mlang here I think. I think the current method has to send all of the objects in the DAO across the network so we can get the length of the array on the client, whereas using a COUNT mlang would only have to send the count across the network. Please correct me if I'm wrong.

Reply:
Nope you're right. I'll make the change